packages/passport/sdk/src/react/PassportContext.tsx

```diff
@@ -0,0 +1,244 @@
+  };
+}
+
+export function useAccessToken() {
+  const { passportInstance, isLoading, isLoggedIn } = usePassport();
+  const [accessToken, setAccessToken] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+
+  const getAccessToken = () => {
+    if (isLoggedIn) {
+      setLoading(true);
+      passportInstance
+        .getAccessToken()
+        .then((t) => setAccessToken(t || null))
+        .catch((e) => setError(e))
+        .finally(() => setLoading(false));
+    } else {
+      setAccessToken(null);
+    }
+
+    return () => {
+      setAccessToken(null);
+    };
+  };
+  useEffect(getAccessToken, [passportInstance, isLoggedIn]);
+  return {
+    accessToken, isLoading: isLoading || loading, error, refetch: getAccessToken,
+  };
+}
+
+export function useLinkedAddresses() {
+  const { passportInstance, isLoading, isLoggedIn } = usePassport();
+  const [linkedAddresses, setLinkedAddresses] = useState<string[] | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+
+  const getLinkedAddresses = () => {
+    if (isLoggedIn) {
+      setLoading(true);
+      passportInstance
+        .getLinkedAddresses()
+        .then((t) => setLinkedAddresses(t || null))
+        .catch((e) => setError(e))
+        .finally(() => setLoading(false));
+    } else {
+      setLinkedAddresses(null);
+    }
+
+    return () => {
+      setLinkedAddresses(null);
+    };
+  };
+  useEffect(getLinkedAddresses, [passportInstance, isLoggedIn]);
+  return {
+    linkedAddresses, isLoading: isLoading || loading, error, refetch: getLinkedAddresses,
+  };
+}
+
+export function useUserInfo() {
+  const { passportInstance, isLoading, isLoggedIn } = usePassport();
+  const [userInfo, setUserInfo] = useState<UserProfile | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+
+  const getUserInfo = () => {
+    if (isLoggedIn) {
+      setLoading(true);
+      passportInstance
+        .getUserInfo()
+        .then((t) => setUserInfo(t || null))
+        .catch((e) => setError(e))
+        .finally(() => setLoading(false));
+    } else {
+      setUserInfo(null);
+    }
+
+    return () => {
+      setUserInfo(null);
+    };
+  };
+  useEffect(getUserInfo, [passportInstance, isLoggedIn]);
+  return {
+    userInfo, isLoading: isLoading || loading, error, refetch: getUserInfo,
+  };
+}
+
+export function useAccounts() {
+  const { accounts, isLoading, error } = usePassport();
+  return { accounts, isLoading, error };
+}
+
+export function usePassportProvider() {
+  const { passportProvider, isLoading, error } = usePassport();
+  return { passportProvider, isLoading, error };
+}
+
+export function useProfile() {
+  const { profile, isLoading, error } = usePassport();
+  return { profile, isLoading, error };
+}
```

Comment:
These three hooks seem redundant if they can just get them from `usePassport`

Reply:
This is mostly for convenience.